### ISSUE TITLE
Things should interact with locations

### ISSUE THREAD

**Issue by incrediblesound** [incrediblesound/story-graph#11]
Things in the world should be able to interact with their locations. It should be possible to make rules that generate events for such interactions.

**Comment by incrediblesound** [comment]
@flipactual  I've been thinking about this for a while. None of my initial ideas seemed to make sense, and  locations being a separate kind of object made interactions between locations and things difficult.

My current idea is that locations should be defined as name-spaced graphs. Currently, there is one graph where all things live and the locations of things are determined by a property on the thing itself. I think it makes more sense to say that the location of a thing is determined by which graph it is in. When you think of it like this everything starts to make sense. For example: the movement of a bird from the forest to the seashore looks like this:

"forest": [Thing("Tree"), Thing("Bird"), Thing("Squirrel")]
"seashore": [Thing("Rock"), Thing("Beach"), Thing("Ocean")]

Transition: Thing("Bird") From "forest" to "seashore" ->

"forest": [Thing("Tree"), Thing("Squirrel")]
"seashore": [Thing("Rock"), Thing("Beach"), Thing("Ocean"), Thing("Bird")]

Now that the bird is in the seashore graph, it can sit on the rock or walk on the beach. I will think more about how to implement this.

**Comment by flipactual** [comment]
I'll have to think on this. Have you checked out other similar projects like https://github.com/galaxykate/tracery/?

**Comment by incrediblesound** [comment]
OK after thinking for a while I think the current implementation is actually half-way there. Having the location tag on each thing is functionally not much different from the example I gave above, and it's much easier to implement. I also improved the location feature a little bit so that it's more flexible. 

I think the only thing we need now is an optional location tag on rules. Certain rules might only apply to certain locations, and so a location property on rules makes sense. There could also be an action constant for something like "maintaining state" or something like that, which when paired with a location property could model rules that render "being in a location". Something like this:

``` javascript
{
  cause: {
    type: [human, c.resting ],
    value: [c.source, 'stares out the window at the falling leaves.' ]
  },
  consequent: {
    type: [],
    value: []
  },
  location: [ 'the kitchen' ],
  isDirectional: false,
  mutations: null,
  consequentThing: null
}
```
